File: Methods&Classes/531.rb
# Write a Song class with attributes and reader/writer methods for name, artist, and duration. Then write a method that prints the name, artist, and duration in a single sentence.

class Song
  attr_reader :name, :artist, :duration
  attr_writer :name, :artist, :duration

  def initialize(name, artist, duration)
    @name = name
    @artist = artist
    @duration = duration
  end

  def sentence
    return "#{name} is a really good song by #{artist}, and is #{duration} minutes long"
  end 
end 

song = Song.new("Fear", "The Score", 2.00)
p song

p song.sentence

File: Methods&Classes/531.js
// Write a Song class with attributes and reader/writer methods for name, artist, and duration. Then write a method that prints the name, artist, and duration in a single sentence.

class Song {
  constructor(name, artist, duration) {
    this.name = name;
    this.artist = artist;
    this.duration = duration;
  }

  sentence() {
    console.log(`The song ${this.name} by ${this.artist} is really good and its ${this.duration} minutes long`);
  }
}

var song = new Song("Survivor", "The Score", 2.13);

console.log(song);

song.sentence();
